datastore/providers/azuresearch_datastore.py

```diff
@@ -0,0 +1,259 @@
+import asyncio
+import os
+import re
+import time
+from typing import Dict, List, Optional, Union
+from datastore.datastore import DataStore
+from models.models import DocumentChunk, DocumentChunkMetadata, DocumentChunkWithScore, DocumentMetadataFilter, Query, QueryResult, QueryWithEmbedding
+from azure.search.documents.aio import SearchClient
+from azure.search.documents.models import Vector, QueryType
+from azure.search.documents.indexes import SearchIndexClient
+from azure.search.documents.indexes.models import *
+from azure.core.credentials import AzureKeyCredential
+from azure.identity import DefaultAzureCredential as DefaultAzureCredentialSync
+from azure.identity.aio import DefaultAzureCredential
+
+AZURESEARCH_SERVICE = os.environ.get("AZURESEARCH_SERVICE")
+AZURESEARCH_INDEX = os.environ.get("AZURESEARCH_INDEX")
+AZURESEARCH_API_KEY = os.environ.get("AZURESEARCH_API_KEY")
+AZURESEARCH_SEMANTIC_CONFIG = os.environ.get("AZURESEARCH_SEMANTIC_CONFIG")
+AZURESEARCH_LANGUAGE = os.environ.get("AZURESEARCH_LANGUAGE", "en-us")
+AZURESEARCH_DISABLE_HYBRID = os.environ.get("AZURESEARCH_DISABLE_HYBRID")
+assert AZURESEARCH_SERVICE is not None
+assert AZURESEARCH_INDEX is not None
+
+# Allow overriding field names for Azure Search
+FIELDS_ID = os.environ.get("AZURESEARCH_FIELDS_ID", "id")
+FIELDS_TEXT = os.environ.get("AZURESEARCH_FIELDS_TEXT", "text")
+FIELDS_EMBEDDING = os.environ.get("AZURESEARCH_FIELDS_TEXT", "embedding")
+FIELDS_DOCUMENT_ID = os.environ.get("AZURESEARCH_FIELDS_DOCUMENT_ID", "document_id")
+FIELDS_SOURCE = os.environ.get("AZURESEARCH_FIELDS_SOURCE", "source")
+FIELDS_SOURCE_ID = os.environ.get("AZURESEARCH_FIELDS_SOURCE_ID", "source_id")
+FIELDS_URL = os.environ.get("AZURESEARCH_FIELDS_URL", "url")
+FIELDS_CREATED_AT = os.environ.get("AZURESEARCH_FIELDS_CREATED_AT", "created_at")
+FIELDS_AUTHOR = os.environ.get("AZURESEARCH_FIELDS_AUTHOR", "author")
+
+# Assume we're using OpenAI's ada-002 embedding model that uses a vector size of 1536
+EMBEDDING_DIMENSIONS = 1536
```

Comment:
not always true, a customer can configure this for Azure Open AI. Please also make this configurable.

Reply:
Done.